engine/src/mac-dialog.mm

```diff
@@ -219,14 +219,28 @@ void MCPlatformBeginFolderDialog(MCPlatformWindowRef p_owner, MCStringRef p_titl
         [t_panel setMessage: [NSString stringWithMCStringRef: p_prompt]];
     }
     else
-        [t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
+    {
+        extern uint4 MCmajorosversion;
+        if (MCmajorosversion >= 0x10B0)
+            [t_panel setMessage: [NSString stringWithMCStringRef: p_prompt]];
+        else
+            [t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
+        
+    }
+        //[t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
```

Comment:
Oops I missed removing the commented-out line.

With this patch, the behavior  is as follows:

OSX < 10.11, all those cases behave as expected. If no title and no prompt are present, you only see an "Open" message in the top of the dialog

OSX >= 10.11:

No title, no promp -> nothing is shown
No title, prompt -> prompt is shown
Title, no prompt -> nothing is shown  (--> this probably needs tweaking)
Title, Prompt -> prompt is shown

So in other words, on OSX >=10.11 with this patch (as with patch https://github.com/livecode/livecode/pull/3789 for the `answer file` case), only the prompt is shown, and the tile is ignored.